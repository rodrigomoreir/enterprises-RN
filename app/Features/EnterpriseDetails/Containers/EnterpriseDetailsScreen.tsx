import React from 'react';
import { useNavigation } from '@react-navigation/native';
import { 
  View, 
  ScrollView,
  Text,
  Image
} from 'react-native';
import { useRoute } from '@react-navigation/core';
import { getStyles } from './EnterpriseDetailsScreen.style'
import { Header } from '../../../Components'
import { EnterpriseProps } from '../../Home/Entities'

interface Params {
  enterprise: EnterpriseProps
}

const EnterpriseDetailsScreen: React.FC = () => {
  const navigation = useNavigation()
  const styles = getStyles()
  const route = useRoute()
  const { enterprise } = route.params as Params

  const navigateToHome = () => navigation.goBack()
  const navigateToUserProfile = () => navigation.navigate('UserProfile')

  return (
    <>
      <Header title={'Detalhes'} goBack={navigateToHome} goProfile={navigateToUserProfile} />
      <ScrollView
        keyboardShouldPersistTaps='handled'
        style={styles.containerScrollView}
        contentContainerStyle={styles.contentScrollView}
      >
        <Text style={styles.title}>
          {enterprise.enterprise_name} - {enterprise.enterprise_type.enterprise_type_name}
        </Text>
        <View style={styles.imageContainer}>
          <Image 
            source={{uri: `https://empresas.ioasys.com.br/${enterprise.photo}`}} 
            style={styles.image} 
          />
        </View>       
        <View style={styles.containerText}>
          <Text style={styles.description}>{enterprise.description}</Text>
          <Text style={styles.location}>{enterprise.city} - {enterprise.country}</Text>
          <View style={styles.contactContainer}>
            <Text style={styles.email}>E-mail: </Text>
            <Text style={styles.textContact}>
              {enterprise.email_enterprise ? 
                `${enterprise.email_enterprise} ðŸ“©` : 'NÃ£o informado. ðŸ˜¢' 
              }
            </Text>
          </View>
          <View style={styles.contactContainer}>
            <Text style={styles.email}>Telefone: </Text>
            <Text style={styles.textContact}>
              {enterprise.email_enterprise ? 
                `${enterprise.phone} ðŸ“²` : 'NÃ£o informado. ðŸ˜¢' 
              }
            </Text>
          </View>
          <View style={styles.contactContainer}>
            <Text style={styles.email}>Facebook: </Text>
            <Text style={styles.textContact}>
              {enterprise.email_enterprise ? 
                `${enterprise.phone} ðŸ“²` : 'NÃ£o informado. ðŸ˜¢' 
              }
            </Text>
          </View>
          <View style={styles.contactContainer}>
            <Text style={styles.email}>Twitter: </Text>
            <Text style={styles.textContact}>
              {enterprise.email_enterprise ? 
                `${enterprise.phone} ðŸ“²` : 'NÃ£o informado. ðŸ˜¢' 
              }
            </Text>
          </View>
          <View style={[styles.contactContainer, {marginBottom: 15}]}>
            <Text style={styles.email}>LinkedIn: </Text>
            <Text style={styles.textContact}>
              {enterprise.email_enterprise ? 
                `${enterprise.phone} ðŸ“²` : 'NÃ£o informado. ðŸ˜¢' 
              }
            </Text>
          </View>
        </View>
      </ScrollView>
    </>
  )
}

export default EnterpriseDetailsScreen;
