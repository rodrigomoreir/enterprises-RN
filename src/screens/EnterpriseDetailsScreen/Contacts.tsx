import React from 'react'

import styled from 'styled-components/native'

interface Props {
  meansOfContact: string
  contact: string
}

const Contacts = ({ meansOfContact, contact }: Props) => {
  return (
    <StyledContainer>
      <StyledTitle>{meansOfContact}</StyledTitle>
      <StyledDescription>{contact}</StyledDescription>
    </StyledContainer>
  )
}

const StyledContainer = styled.View`
  flex-direction: row;
  align-items: flex-end;
`
const StyledTitle = styled.Text`
  margin-top: 10px;
  font-size: 18px;
  font-style: normal;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.darkGrey};
  align-self: flex-start;
`

const StyledDescription = styled.Text`
  font-size: 15px;
  line-height: 22px;
  font-style: normal;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.darkGrey};
`

export default Contacts
