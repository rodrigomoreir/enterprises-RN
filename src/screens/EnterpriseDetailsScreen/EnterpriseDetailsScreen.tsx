import React from 'react'
import { useNavigation } from '@react-navigation/native'
import { SafeAreaView } from 'react-native'

import { useRoute } from '@react-navigation/core'

import { Header } from '../../core'
import { EnterpriseProps } from '../HomeScreen/Entities'
import Contacts from './Contacts'
import { screenWidth } from '../../utils'

import styled from 'styled-components/native'

interface Params {
  enterprise: EnterpriseProps
}

const EnterpriseDetailsScreen: React.FC = () => {
  const navigation = useNavigation()
  const route = useRoute()
  const { enterprise } = route.params as Params

  const navigateToHome = () => navigation.goBack()
  const navigateToUserProfile = () => navigation.navigate('UserProfile')

  console.log('EnterpriseDetails rendered')

  return (
    <SafeAreaView>
      <Header
        title={'Detalhes'}
        goBack={navigateToHome}
        goProfile={navigateToUserProfile}
      />
      <StyledScrollView keyboardShouldPersistTaps="handled" bounces={false}>
        <StyledTitle>
          {enterprise.enterprise_name} -{' '}
          {enterprise.enterprise_type.enterprise_type_name}
        </StyledTitle>
        <StyledViewImage>
          <StyledImage
            source={{
              uri: `https://empresas.ioasys.com.br/${enterprise.photo}`,
            }}
          />
        </StyledViewImage>
        <StyledEnterpriseDescription>
          <StyledTextDescription>
            {enterprise.description}
          </StyledTextDescription>
          <StyledTextLocation>
            {enterprise.city} - {enterprise.country}
          </StyledTextLocation>
          <Contacts
            meansOfContact={'E-mail: '}
            contact={
              enterprise.email_enterprise
                ? `${enterprise.email_enterprise}`
                : 'NÃ£o informado. ðŸ˜¢'
            }
          />
          <Contacts
            meansOfContact={'Telefone: '}
            contact={
              enterprise.phone ? `${enterprise.phone}` : 'NÃ£o informado. ðŸ˜¢'
            }
          />
          <Contacts
            meansOfContact={'Facebook: '}
            contact={
              enterprise.facebook
                ? `${enterprise.facebook}`
                : 'NÃ£o informado. ðŸ˜¢'
            }
          />
          <Contacts
            meansOfContact={'Twitter: '}
            contact={
              enterprise.twitter ? `${enterprise.twitter}` : 'NÃ£o informado. ðŸ˜¢'
            }
          />
          <Contacts
            meansOfContact={'LinkedIn: '}
            contact={
              enterprise.linkedin
                ? `${enterprise.linkedin}`
                : 'NÃ£o informado. ðŸ˜¢'
            }
          />
        </StyledEnterpriseDescription>
      </StyledScrollView>
    </SafeAreaView>
  )
}

const StyledScrollView = styled.ScrollView.attrs({
  contentContainerStyle: {
    flexGrow: 1,
    width: screenWidth,
    alignItems: 'center',
    justifyContent: 'space-between',
  },
})`
  width: ${screenWidth}px;
`
const StyledTitle = styled.Text`
  margin-top: 10px;
  font-size: 20px;
  line-height: 26.38px;
  font-style: normal;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.darkGrey};
  text-align: center;
`
const StyledViewImage = styled.View`
  margin-top: 10px;
  width: ${screenWidth - 30}px;
  height: 250px;
  border-radius: 20px;
`

const StyledImage = styled.Image`
  width: 100%;
  height: 100%;
  border-radius: 20px;
`

const StyledEnterpriseDescription = styled.View`
  flex: 1;
  width: ${screenWidth - 33}px;
  align-items: center;
`
const StyledTextDescription = styled.Text`
  margin-top: 10px;
  font-size: 15px;
  line-height: 22px;
  font-style: normal;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.darkGrey};
  text-align: justify;
`
const StyledTextLocation = styled.Text`
  margin-bottom: 10px;
  font-size: 15px;
  line-height: 22px;
  font-style: normal;
  color: ${({ theme }) => theme.colors.darkGrey};
  align-self: center;
`

export default EnterpriseDetailsScreen
