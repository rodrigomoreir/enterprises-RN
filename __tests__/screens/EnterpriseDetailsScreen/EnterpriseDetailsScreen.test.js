import React from 'react'
import EnterpriseDetailsScreen from '../../../src/screens/EnterpriseDetailsScreen/EnterpriseDetailsScreen'

import {
  renderWithTheme,
  matchSnapshotWithTheme,
} from '../../../src/functions/testing'

describe('Enterprise Details Screen', () => {
  test('matches snapshot', () => {
    matchSnapshotWithTheme(<EnterpriseDetailsScreen />)
  })

  test('renders correctly', () => {
    renderWithTheme(<EnterpriseDetailsScreen />)
  })
})
