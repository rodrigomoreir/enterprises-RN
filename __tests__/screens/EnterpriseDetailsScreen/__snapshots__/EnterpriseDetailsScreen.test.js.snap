// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enterprise Details Screen matches snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomWidth": 2,
          "borderColor": "#a9a9a9",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "paddingHorizontal": 10,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "width": 40,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#87cefaCC",
            "borderColor": "#69696999",
            "borderRadius": 20,
            "borderWidth": 1,
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "width": 40,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/chevronLeft.png",
            }
          }
          style={
            Array [
              Object {
                "height": 20,
                "tintColor": "#ecf0f2",
                "width": 20,
              },
            ]
          }
        />
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#696969",
            "fontSize": 20,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "lineHeight": 26.38,
            "textAlign": "center",
          },
        ]
      }
    >
      Detalhes
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#69696999",
          "borderRadius": 20,
          "borderWidth": 1,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 40,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/profile.png",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 20,
              "maxHeight": 38,
              "maxWidth": 38,
              "tintColor": "#87cefaCC",
            },
          ]
        }
      />
    </View>
  </View>
  <RCTScrollView
    bounces={false}
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flexGrow": 1,
        "justifyContent": "space-between",
        "width": 750,
      }
    }
    keyboardShouldPersistTaps="handled"
    style={
      Array [
        Object {
          "width": 750,
        },
      ]
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#696969",
              "fontSize": 20,
              "fontStyle": "normal",
              "fontWeight": "bold",
              "lineHeight": 26.38,
              "marginTop": 10,
              "textAlign": "center",
            },
          ]
        }
      >
         -
         
      </Text>
      <View
        style={
          Array [
            Object {
              "borderRadius": 20,
              "height": 250,
              "marginTop": 10,
              "width": 720,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "https://empresas.ioasys.com.br/undefined",
            }
          }
          style={
            Array [
              Object {
                "borderRadius": 20,
                "height": "100%",
                "width": "100%",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "width": 717,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#696969",
                "fontSize": 15,
                "fontStyle": "normal",
                "fontWeight": "bold",
                "lineHeight": 22,
                "marginTop": 10,
                "textAlign": "justify",
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#696969",
                "fontSize": 15,
                "fontStyle": "normal",
                "lineHeight": 22,
                "marginBottom": 10,
              },
            ]
          }
        >
           - 
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-end",
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#696969",
                  "fontSize": 18,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "marginTop": 10,
                },
              ]
            }
          >
            E-mail: 
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#696969",
                  "fontSize": 15,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "lineHeight": 22,
                },
              ]
            }
          >
            NÃ£o informado. ðŸ˜¢
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-end",
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#696969",
                  "fontSize": 18,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "marginTop": 10,
                },
              ]
            }
          >
            Telefone: 
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#696969",
                  "fontSize": 15,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "lineHeight": 22,
                },
              ]
            }
          >
            NÃ£o informado. ðŸ˜¢
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-end",
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#696969",
                  "fontSize": 18,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "marginTop": 10,
                },
              ]
            }
          >
            Facebook: 
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#696969",
                  "fontSize": 15,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "lineHeight": 22,
                },
              ]
            }
          >
            NÃ£o informado. ðŸ˜¢
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-end",
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#696969",
                  "fontSize": 18,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "marginTop": 10,
                },
              ]
            }
          >
            Twitter: 
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#696969",
                  "fontSize": 15,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "lineHeight": 22,
                },
              ]
            }
          >
            NÃ£o informado. ðŸ˜¢
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-end",
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#696969",
                  "fontSize": 18,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "marginTop": 10,
                },
              ]
            }
          >
            LinkedIn: 
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#696969",
                  "fontSize": 15,
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "lineHeight": 22,
                },
              ]
            }
          >
            NÃ£o informado. ðŸ˜¢
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
